src/dclass.d

```diff
@@ -470,20 +470,32 @@ public:
 
         if (baseok < BASEOKdone)
         {
+            /* Bugzilla 12078, 12143 and 15733:
+             * While resolving base classes and interfaces, a base may refer
+             * the member of this derived class. In that time, if all bases of
+             * this class can  be determined, we can go forward the semantc process
+             * beyond the Lancestorsdone. To do the recursive semantic analysis,
+             * temporarily set and unset `_scope` around exp().
+             */
+            T resolveBase(T)(lazy T exp)
+            {
+                if (!scx)
+                {
+                    scx = sc.copy();
+                    scx.setNoFree();
+                }
+                _scope = scx;
+                scope (exit) _scope = null;
```

Comment:
Using scope exit involves creating an exception handling frame. However, none of dmd throws exceptions, making this inefficient. Just set _scope to null after the call to exp().